router.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package ringpop
+
+import (
+	"sync"
+
+	"github.com/uber/ringpop-go/swim"
+)
+
+// Replication constants
+type ReplicationFactor int
+
+const (
+	ALL ReplicationFactor = -1 * (iota + 1)
```

Comment:
That is a good point that I have been thinking about as well. golang does not allow you to specify specific defaults for fields if they are not specified and initializes all fields to the zero value of the type. I was thinking about validating the configuration for all endpoints in the construction of the Adapter. So in this validation step we can check if `N` and `Fanout` are `0` and set them to the sane default of 1. In that case the configuration is very small and looks like this:

```go
&KeyValueGetConfiguration{
    Key: func(ctx thrift.Context, key string) (string,error) {
        return key, nil
    }
}
```

And it will work. Although since in the example from above `N` is set to 2 you still need to set that on this configuration.